streams/src/main/java/org/apache/kafka/streams/processor/AbstractNotifyingBatchingRestoreCallback.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.processor;
+
+
+import org.apache.kafka.common.TopicPartition;
+
+/**
+ * Abstract implementation of the  {@link BatchingStateRestoreCallback} used for batch restoration operations.
+ *
+ * Includes default no-op methods of the {@link StateRestoreListener} {@link StateRestoreListener#onRestoreStart(TopicPartition, String, long, long)},
+ * {@link StateRestoreListener#onBatchRestored(TopicPartition, String, long, long)}, and {@link StateRestoreListener#onRestoreEnd(TopicPartition, String, long)}.
+ */
+public abstract class AbstractNotifyingBatchingRestoreCallback
+    implements BatchingStateRestoreCallback, StateRestoreListener {
+
+    /**
+     * Single put restore operations not supported, please use {@link AbstractNotifyingRestoreCallback}
+     * or {@link StateRestoreCallback} instead for single action restores.
+     */
+    @Override
+    public void restore(byte[] key, byte[] value) {
```

Comment:
Can we introduce a global "one parameter per line" code style? I think it would help to make diffs cleaner. We can do this incrementally. If yes, please do for all newly introduced code of this PR.

Also, should be add `final` all over the place?